server/models/User.ts

```diff
@@ -6,26 +6,27 @@ class User extends Model {}
 User.init(
   {
     // Model attributes are defined here
-    id: {
-      type: DataTypes.UUID,
-      primaryKey: true,
-      defaultValue: DataTypes.UUIDV4,
-    },
     username: {
       type: DataTypes.STRING,
+      unique: true,
       allowNull: false,
     },
     email: {
       type: DataTypes.STRING,
       allowNull: false,
+      unique: true,
       validate: {
         isEmail: true,
       },
     },
-    password: {
+    passwordHash: {
       type: DataTypes.STRING,
```

Comment:
Can we have a custom setter, which directly hashes the password: https://sequelize.org/docs/v6/core-concepts/getters-setters-virtuals/#setters